packages/core/src/components/forms/controls.tsx

```diff
@@ -129,13 +131,43 @@ const Control: React.SFC<IControlInternalProps> = ({
 // Switch
 //
 
-export interface ISwitchProps extends IControlProps {}
+export interface ISwitchProps extends IControlProps {
+    /**
+     * String to display within the switch control component when the switch is active ("on").
+     */
+    internalTextActive?: string;
+
+    /**
+     * String to display within the switch control component when the switch is inactive ("off").
+     */
+    internalTextInactive?: string;
+}
 
 export class Switch extends React.PureComponent<ISwitchProps> {
     public static displayName = `${DISPLAYNAME_PREFIX}.Switch`;
-
+    private switchText = [
```

Comment:
what happens if one of the text props is omitted?

Reply:
also this should be defined in `render`, not at construction time as a member (where it will never be updated to reflect props changes)

Reply:
oops :-P